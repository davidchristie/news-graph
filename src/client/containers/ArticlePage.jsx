import React, { PropTypes } from 'react'
import { connect } from 'react-redux'

// import ArticleNetwork from './ArticleNetwork'
import Jumbotron from '../components/Jumbotron'

export const ArticlePage = class ArticlePage extends React.Component {
  render () {
    const url = this.props.location.query.url
    const articles = this.props.articles
    const title = 'Title'
    const description = 'Description'
    // const article = articles[url]
    // const title = article.title
    // const description = article.description
    // <ArticleNetwork article={article} />
    return (
      <div>
        <Jumbotron title={title} lead={description} />
        <div className='card'>
          <div className='card-block'>
            <h2 className='card-title'>Connections</h2>
          </div>
        </div>
      </div>
    )
  }
}

ArticlePage.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }).isRequired).isRequired,
  location: PropTypes.shape({
    query: PropTypes.shape({
      url: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
}

function mapStateToProps (state) {
  return {
    articles: state.app.articles
  }
}

export default connect(mapStateToProps)(ArticlePage)
