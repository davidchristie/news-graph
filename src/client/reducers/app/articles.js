// Article:
// {
//   author
//   description
//   publishedAt
//   title
//   url
//   urlToImage
// }

export default (state = [], action) => {
  switch (action.type) {
    case 'ADD_ARTICLE':
      return [...state, action.article]
    default:
      return state
  }
}
