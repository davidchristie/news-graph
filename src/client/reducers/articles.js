export default (state = [], action) => {
  switch (action.type) {
    case 'ADD_ARTICLE':
      return [...state, action.article]
    default:
      return state
  }
}
