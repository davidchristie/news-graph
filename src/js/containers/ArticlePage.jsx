import React, { PropTypes } from 'react'

import ArticleNetwork from './ArticleNetwork'

export default class ArticlePage extends React.Component {
  render() {
    const id = this.props.params.id
    return (
      <div className="container">
        <h1>ArticlePage ({id})</h1>
        <div className="row">
          <ArticleNetwork />
        </div>
      </div>
    )
  }
}

ArticlePage.propTypes = {
  params: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired
}
