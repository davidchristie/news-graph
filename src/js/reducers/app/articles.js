let nextId = 1

// Article:
// {
//   author
//   description
//   publishedAt
//   title
//   url
//   urlToImage
// }

function createArticle(article) {
  return Object.assign(article, {id: nextId++})
}

export default (state=[], action) => {
  switch (action.type) {
  case 'ADD_ARTICLE':
    return [...state, createArticle(action.article)]
  case 'POST_ARTICLE': // FIXME Needs to fetch article title, urlToImage etc
    return [...state, {
      id: nextId++,
      url: action.url
    }]
  default:
    return state
  }
}
