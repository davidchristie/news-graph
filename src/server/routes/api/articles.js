const express = require('express')

const api = require('../../api')

const router = express.Router()

router.get('/', (request, response) => {
  api.getArticles()
    .then(articles => response.json({articles}))
    .catch(error => response.send(error.message))
})

module.exports = router
