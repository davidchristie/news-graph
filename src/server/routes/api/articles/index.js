const express = require('express')

const neo4j = require('../../../databases/neo4j')
// const newsapi = require('../../../externals/newsapi')

const router = express.Router()

router.get('/featured', (request, response) => {
  return neo4j.getRecentArticles()
    .then(articles => response.json({articles}))
    .catch(error => response.send(error.message))

  // newsapi.getArticles()
  //   .then(articles => response.json({articles}))
  //   .catch(error => response.send(error.message))
})

module.exports = router
