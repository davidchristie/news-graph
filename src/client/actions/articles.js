import { post } from '../api'

let nextId = 1

// TODO Remove this
export function addArticle (article) {
  return {
    article: Object.assign({}, article, {id: nextId++}),
    type: 'ADD_ARTICLE'
  }
}

export function addConnection (name, from, to) {
  return {
    from: from,
    name: name,
    to: to,
    type: 'ADD_CONNECTION'
  }
}

export function postArticle ({ token, url }) {
  return dispatch => {
    dispatch(postArticlePending(url))
    return post('articles', {token, url})
      .then(response => response.data)
      .then(data => {
        if (data.success) {
          return dispatch(postArticleSuccess({url}))
        } else {
          return dispatch(postArticleFailure({
            url,
            message: data.message
          }))
        }
      })
      .catch(error => {
        return dispatch(postArticleFailure({
          alerts: [
            {
              text: error.message,
              type: 'danger'
            }
          ],
          message: error.message,
          url
        }))
      })
  }
}

function postArticlePending ({ url }) {
  return {
    type: 'POST_ARTICLE_PENDING',
    url
  }
}

function postArticleSuccess ({ url }) {
  return {
    type: 'POST_ARTICLE_SUCCESS',
    url
  }
}

function postArticleFailure ({ url, message }) {
  return {
    message,
    type: 'POST_ARTICLE_FAILURE',
    url
  }
}
