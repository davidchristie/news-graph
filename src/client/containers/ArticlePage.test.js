/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import articles from '../examples/articles'
import { ArticlePage } from './ArticlePage'

test('ArticlePage renders correctly', () => {
  const query = {url: articles[0].url}
  const location = {query}
  const wrapper = shallow(<ArticlePage articles={articles} location={location} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
