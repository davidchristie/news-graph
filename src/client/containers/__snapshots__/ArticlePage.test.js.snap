exports[`test ArticlePage renders correctly 1`] = `
<div>
  <Jumbotron
    lead="Description"
    title="Title" />
  <div
    className="card">
    <div
      className="card-block">
      <h2
        className="card-title">
        Connections
      </h2>
    </div>
  </div>
</div>
`;
